File: loja/pedidos/src/main/java/br/com/bota/ws/filter/ClientFilter.java
package br.com.bota.ws.filter;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.MutableHttpRequest;
import io.micronaut.http.annotation.Filter;
import io.micronaut.http.filter.ClientFilterChain;
import io.micronaut.http.filter.HttpClientFilter;
import org.reactivestreams.Publisher;

import java.util.UUID;

@Filter("**")//pattern
public class ClientFilter implements HttpClientFilter {
    @Override
    public Publisher<? extends HttpResponse<?>> doFilter(MutableHttpRequest<?> request, ClientFilterChain chain) {
        return chain.proceed(request.header("correlationId", UUID.randomUUID().toString()));
    }
}

File: loja-kotlin/kpedidos/src/main/kotlin/br/com/bota/ws/filter/ClientFilter.kt
package br.com.bota.ws.filter

import io.micronaut.http.HttpResponse
import io.micronaut.http.MutableHttpRequest
import io.micronaut.http.annotation.Filter
import io.micronaut.http.filter.ClientFilterChain
import io.micronaut.http.filter.HttpClientFilter
import org.reactivestreams.Publisher
import java.util.*

@Filter("**")
class ClientFilter : HttpClientFilter {
    override fun doFilter(request: MutableHttpRequest<*>, chain: ClientFilterChain): Publisher<out HttpResponse<*>> {
        return chain.proceed(request.header("correlationId", UUID.randomUUID().toString()))
    }
}